code/modules/research/xenobiology/xenobiology_dissections.dm

```diff
@@ -0,0 +1,1533 @@
+	. = ..()
+	var/datum/spell/head_attack/spell = new
+	spell.quality = organ_quality
+	if(organ_quality == ORGAN_PRISTINE)
+		spell.base_cooldown = 3 MINUTES
+	M.AddSpell(spell)
+
+/obj/item/organ/internal/ears/xenobiology/sinister/remove(mob/living/carbon/M, special = 0)
+	. = ..()
+	M.RemoveSpell(/datum/spell/head_attack)
+
+/datum/spell/head_attack
+	name = "Aggressive Budding"
+	desc = "Grow and detach a new head from yourself to send towards your enemies. Cast again to destroy any existing heads."
+	base_cooldown = 5 MINUTES
+	clothes_req = FALSE
+	stat_allowed = CONSCIOUS
+	invocation_type = "*scream"
+	action_icon = 'icons/mob/actions/actions_elites.dmi'
+	action_icon_state = "head_detach"
+	action_background_icon_state = "bg_default"
+	sound = 'sound/effects/blood1.ogg'
+	sound = null
+	active = FALSE
+	var/quality
+	var/mob/living/simple_animal/hostile/asteroid/elite/legionnairehead/xenobiology/newhead
+
+/datum/spell/head_attack/create_new_targeting(list/targets, mob/living/user)
+	. = ..()
+	var/datum/spell_targeting/clicked_atom/external/C = new()
+	C.range = 20
+	return C
+
+/datum/spell/head_attack/Click(mob/user)
+	. = ..()
+	if(newhead)
+		to_chat(user, "<span class='notice'>We disperse the summoned head.</span>")
+		on_head_death()
+		return
+
+/datum/spell/head_attack/cast(list/targets, mob/living/carbon/human/user)
+	. = ..()
+	if(!isliving(targets[1]))
+		to_chat(user, "<span class='warning'>We can only select living targets!</span>")
+		revert_cast()
+		return
+	var/mob/living/target = targets[1] // only ever one target
+	if(!user.get_organ("head"))
+		to_chat(user, "<span class='warning'>We cant use this without a head!</span>")
+		revert_cast()
+		return
+	if(target == user)
+		to_chat(user, "<span class='warning'>It would be a bad idea to target ourself.</span>")
+		revert_cast()
+		return
+	if(target.stat == DEAD)
+		to_chat(user, "<span class='warning'>Our new head wont attack corpses</span>")
+		revert_cast()
+		return
+	if(!newhead)
+		if(quality == ORGAN_DAMAGED && prob(30))
+			user.adjustBruteLoss(15) // ouch!
+		user.visible_message("<span class='warning'>[src] produces a new head!</span>", "<span class='warning'>You produce a new head and send it to your target!</span>")
+		newhead = new /mob/living/simple_animal/hostile/asteroid/elite/legionnairehead/xenobiology(user.loc)
+		newhead.GiveTarget(target)
+		newhead.faction = user.faction.Copy()
+		newhead.icon = null
+		newhead.name = "[user]'s head"
+		newhead.parent_spell = src
+		newhead.faction = list(user)
+		newhead.permanent_target = target
+		var/obj/item/user_head = user.get_organ("head")
+		var/obj/item/dupe_head = DuplicateObject(user_head, perfectcopy=1, sameloc=0, newloc=newhead.contents)
+		newhead.mounted_head = dupe_head
+		var/matrix/M = matrix()
+		dupe_head.transform = M
+		var/image/IM = image(dupe_head.icon, dupe_head.icon_state)
+		IM.pixel_y -= 6
+		IM.overlays = dupe_head.overlays.Copy()
+		newhead.overlays += IM
```

Comment:
Could you move all of this logic into the Initialize() of the `.../legionnairehead/xenobiology` from the user and target?